Autopilot/Inc/timeStampingPOGI.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef TIME_STAMPING_POGI_HPP
+#define TIME_STAMPING_POGI_HPP
+
+#include <cstdint>
+#include <cstdlib>
+#include "gps.hpp"
+
+class TimeStampingPOGI {
+
+    public:
+
+        /*
+        *Called to either create or get the existing instance of the class
+        *Constructor is private so it only gets called through here (singleton)
+        */
+        static TimeStampingPOGI* GetInstance();
```

Comment:
Need a line deleting the constructor so people don't accidentally initialize it. You can look at line 67 of this file: https://github.com/UWARG/ZeroPilot-SW/blob/devel/Autopilot/Inc/gps.hpp